src/components/Question/Question.tsx

```diff
@@ -1,36 +1,67 @@
-import React, { useState, useContext}  from 'react';
+import React, { useState, useContext, useEffect}  from 'react';
+import { Link } from 'react-router-dom';
 import {QuestionContext, AnswerContext} from '../../types';
 import './Question.css'
 import bkg_img from '../../images/questions/Big Shoes - Sitting On Floor.png'
 import location_img from '../../images/questions/Charco - Location Map.png'
 
 interface Props{
   updateAllAnswers: any;
-  currentQuestion: string;
 }
-    export const Question: React.FC<Props> = ({updateAllAnswers, currentQuestion}) => {
+    export const Question: React.FC<Props> = ({updateAllAnswers}) => {
 
     const answerContext = useContext(AnswerContext)
     const questionContext = useContext(QuestionContext)
-    const [answerInput, updateAnswer] = useState<any>({question: '', answer: ''});
+    const questionSet = Object.keys(questionContext);
 
-    let currentAnswer = Object.keys(answerContext).find(question => answerContext[question])
-    let theQuestion = questionContext[currentQuestion]
+    //local state
+    const [answerInput, updateAnswer] = useState<any>({});
+    const [index, setIndex] = useState<number>(0);
+    // const[error, setError] = useState<boolean>(false);
+    const [errorMessage, setErrorMessage] = useState<string>('');
 
-    console.log(theQuestion)
+    let currentQuestion = questionContext[questionSet[index]]
+
+    // const nextQuestion = () => {
+    //   console.log(answerInput[questionSet[index]])
+    //   if(!answerInput[questionSet[index]]) {
+    //     setError(true);
+    //     setErrorMessage('Sorry but we need this information')
+    //     setTimeout(()=> setError(false), 2500)
+    //   } else if(index < questionSet.length && answerInput[questionSet[index]]) {
+    //     setIndex(prevIndex => prevIndex + 1);
+    //     setError(false)
+    //   }
+    // }
+
+    const nextQuestion = () => {
```

Comment:
still wondering how we can move these set states out of nextQuestion. I think a custom hook may make the most sense if we have time

Reply:
I completely agree